dev/planning-example.py

```diff
@@ -0,0 +1,236 @@
+"""
+
+# Define a time
+from astropy.time import Time
+time_obs = Time(2457189.500000, format='jd')
+
+# sunset
+obs.sunset(date=time_obs, which='nearest') # Default
+obs.sunset(date=time_obs, which='next')
+obs.sunset(date=time_obs, which='previous')
+
+# sunrise
+obs.sunrise(date=time_obs, which='nearest') # Default
+obs.sunrise(date=time_obs, which='next')
+obs.sunrise(date=time_obs, which='previous')
+
+# moon rise
+obs.moon_rise(date=time_obs, which='nearest')
+
+# moon set
+obs.moon_set(date=time_obs, which='nearest')
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight.
+# For example, to compute astronomical twilight by specifying the `angle`:
+obs.sunset(date=time_obs, which='next', angle=18*u.degree) 
+
+# evening (astronomical) twilight
+obs.evening_astronomical(date=time_obs)
+
+# evening (nautical) twilight
+obs.evening_nautical(date=time_obs)
+
+# evening (civil) twilight
+obs.evening_civil(date=time_obs)
+
+# morning (nautical) twilight
+obs.morning_nautical(date=time_obs)
+
+# morning (civil) twilight
+obs.morning_civil(date=time_obs)
+
+# morning (astronomical) twilight
+obs.morning_astronomical(date=time_obs)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(date=time_obs)
+
+# what is the moon altitude and azimuth?
+obs.moon_altaz(date=time_obs)
+
+# Other sun-related convenience functions:
+obs.noon(date=time_obs, which='nearest')
+obs.midnight(date=time_obs, which='nearest')
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan import FixedTarget, airmass_plot
+
+# Define a target.
+t1 = FixedTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = FixedTarget(name='Polaris',
+               coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery:
+t1 = FixedTarget.from_name('Polaris')
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST, above airmass 1.2, on a telescope that can observe between 
+15 degrees and 89 degrees altitude?
+"""
+
+# first we define the conditions
+from astroplan import TimeWindow, AltitudeWindow, AboveAirmass, is_observable
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+constraint_list = [TimeWindow("2015-05-01 18:30", "2015-05-02 05:30"),
```

Comment:
@eteq @cdeil I'm about to change the time boundaries accepted by `TimeWindow` be instances of `astropy.time.Time` rather than strings. Sound good?